analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.LearnConstraint.cs

```diff
@@ -433,6 +435,76 @@ public void Branching_LearnsObjectConstraint_TypePattern_Negated(string expressi
                 .And.ContainSingle(x => x != null && x.HasConstraint(ObjectConstraint.NotNull));
         }
 
+        [DataTestMethod]
+        [DataRow("arg is { }")]
+        [DataRow("arg is { }", "TClass")]
+        [DataRow("arg is object { }")]
+        [DataRow("arg is not not { }")]
+        [DataRow("!!(arg is { })")]
+        [DataRow("arg is (A: var a, B: _)", "Deconstructable")]
+        public void Branching_LearnsObjectConstraint_RecursivePattern_ElseIsNull(string expression, string argType = "object")
+        {
+            var validator = CreateIfElseEndValidatorCS(expression, OperationKind.RecursivePattern, argType);
+            validator.ValidateTag("If", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.ValidateTag("Else", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
+            validator.TagValues("End").Should().HaveCount(2)
+                .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.Null))
+                .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.NotNull));
+        }
+
+        [DataTestMethod]
+        [DataRow("arg is string { }")]
+        [DataRow("arg is string { Length: 0 }")]
+        [DataRow("arg is string { Length: var length }")]
```

Comment:
```suggestion
        [DataRow("arg is string { Length: var length }")]
        [DataRow("arg is int { }")]
        [DataRow("arg is DateTime { Ticks: 0 }")]
        [DataRow("arg is TClass { }")]
        [DataRow("arg is T { }")]
        [DataRow("arg is TStruct { }")]
        [DataRow("arg is string { }", "TClass")]
        [DataRow("arg is string { }", "T")]        
```

Reply:
Except `[DataRow("arg is string { }", "T")]` that didn't work (will not fix)

Reply:
What about `[DataRow("arg is string { }", "TClass")]`? It is missing too.

Reply:
Line 464